splinepy/io/gismo.py

```diff
@@ -302,13 +311,45 @@ def export(
     patch_range = _ET.SubElement(
         multipatch_element, "patches", type="id_range"
     )
-    patch_range.text = (
-        f"{index_offset} " f"{len(multipatch.patches) - 1 + index_offset}"
+
+    ###
+    # Individual spline data
+    ###
+    # Export fields first, as all necessary information is already available
+    if export_fields:
+        field_xml = _copy.deepcopy(xml_data)
+
+        n_patches = _spline_to_ET(
+            field_xml,
+            multipatch,
+            index_offset,
+            fields_only=True,
+            as_base64=as_base64,
+            field_mask=field_mask,
+        )
+        field_xml.find("MultiPatch").find("patches").text = (
+            f"{index_offset} " f"{n_patches + index_offset}"
+        )
+        if int(_python_version.split(".")[1]) >= 9 and indent:
+            _ET.indent(field_xml)
+        file_content = _ET.tostring(field_xml)
+        with open(fname + ".fields.xml", "wb") as f:
+            f.write(file_content)
+
+    n_patches = _spline_to_ET(
```

Comment:
This Block of Code must stay where it was otherwise the order will change

Reply:
It's successful now, thanks!